onnxruntime/test/shared_lib/test_inference.cc

```diff
@@ -357,7 +370,17 @@ TEST(CApiTest, test_custom_op_library) {
 lib_name = "./libcustom_op_library.so";
 #endif
 
-  TestInference<PATH_TYPE, int32_t>(*ort_env, CUSTOM_OP_LIBRARY_TEST_MODEL_URI, inputs, "output", expected_dims_y, expected_values_y, 0, nullptr, lib_name.c_str());
+  void* library_handle = nullptr;
```

Comment:
Here we are unloading it but not in other cases. Why not?

Reply:
Because this is the only test that actually needs an unload. This is the only test that actually loads the library.

Reply:
Simplified everything - should be much cleaner now